src/JSInterop/Microsoft.JSInterop.JS/src/src/Microsoft.JSInterop.ts

```diff
@@ -294,17 +300,19 @@ export module DotNet {
      *
      * @param identifier Identifies the globally-reachable function to invoke.
      * @param argsJson JSON representation of arguments to be passed to the function.
+     * @param byteArrays Byte array data extracted from the arguments for direct transfer.
      * @param resultType The type of result expected from the JS interop call.
      * @param targetInstanceId The instance ID of the target JS object.
      * @returns JSON representation of the invocation result.
      */
-    invokeJSFromDotNet: (identifier: string, argsJson: string, resultType: JSCallResultType, targetInstanceId: number) => {
-      const returnValue = findJSFunction(identifier, targetInstanceId).apply(null, parseJsonWithRevivers(argsJson));
+    invokeJSFromDotNet: (identifier: string, argsJson: string, byteArrays: Uint8Array[], resultType: JSCallResultType, targetInstanceId: number) => {
+      const returnValue = findJSFunction(identifier, targetInstanceId).apply(null, parseJsonWithRevivers(argsJson, byteArrays));
       const result = createJSCallResult(returnValue, resultType);
 
+      const returnedByteArrays : Uint8Array[] = [];
       return result === null || result === undefined
         ? null
-        : JSON.stringify(result, argReplacer);
+        : JSON.stringify(result, (key, value) => argReplacer(key, value, returnedByteArrays)); // TODO; what is this used for? Seems like it doesn't actually invoke anything
```

Comment:
Haven't been able to test this out, do we actually do a synchronous JS invocation from dotnet?

Reply:
Blazor server doesn't support synchronous interop (all calls go through the network and that's asynchronous by definition)

Reply:
@TanayParikh You probably already know this, but just to be sure: Blazor WebAssembly *does* support synchronous interop in both directions.